src/api/app/models/bs_request_action.rb

```diff
@@ -74,24 +74,7 @@ def self.get_package_diff(path, query)
   end
 
   def self.type_to_class_name(type_name)
-    case type_name
-    when :submit then
-      BsRequestActionSubmit
-    when :delete then
-      BsRequestActionDelete
-    when :change_devel then
-      BsRequestActionChangeDevel
-    when :add_role then
-      BsRequestActionAddRole
-    when :set_bugowner then
-      BsRequestActionSetBugowner
-    when :maintenance_incident then
-      BsRequestActionMaintenanceIncident
-    when :maintenance_release then
-      BsRequestActionMaintenanceRelease
-    when :group then
-      BsRequestActionGroup
-    end
+    ApplicationController.helpers.model_name_to_class("bs_request_action_#{type_name}")
```

Comment:
Why are you creating a helper and using it in the model? If instead of this helper we use something like:
```
"bs_request_action_#{type_name}".classify.contantize
```